tests/unit/executor/test_executor.py

```diff
@@ -199,6 +203,8 @@ def test_prologue_overlay_packages(self, new_dir, mocker):
         with ExecutionContext(executor=e):
             assert not mock_mount.called
 
+        Features.reset()
```

Comment:
this is tricky because if the test fails before this line is reached it will never be called, which might make _other_ tests that come after this fail and that's usually very annoying to debug.

You could wrap the whole thing in a try/finally but I think the `request` fixture would be cleaner:

```python
    def test_prologue_overlay_packages(self, new_dir, mocker, request):
        """Check that the overlay package cache is not touched if the part doesn't have overlay packages"""
        Features.reset()
        Features(enable_overlay=True)
        request.addfinalizer(Features.reset)

Reply:
Done as suggested.